pkg/u8g2/patches/0002-u8g2-add-riot-os-interface.patch

```diff
@@ -0,0 +1,237 @@
++        case U8X8_MSG_BYTE_SEND:
++            spi_transfer_bytes(dev, (char *) arg_ptr, NULL, arg_int);
++            break;
++        case U8X8_MSG_BYTE_INIT:
++            spi_init_master(dev, 
++                            u8x8_takeover_edge_to_spi_conf(u8g2->display_info->sck_takeover_edge), 
++                            u8x8_pulse_width_to_spi_speed(u8g2->display_info->sck_pulse_width_ns));
++            break;
++        case U8X8_MSG_BYTE_SET_DC:
++            u8x8_gpio_SetDC(u8g2, arg_int);
++            break;
++        case U8X8_MSG_BYTE_START_TRANSFER:
++            spi_acquire(dev);
++
++            u8x8_gpio_SetCS(u8g2, u8g2->display_info->chip_enable_level);
++            u8g2->gpio_and_delay_cb(u8g2, U8X8_MSG_DELAY_NANO, u8g2->display_info->post_chip_enable_wait_ns, NULL);
++            break;
++        case U8X8_MSG_BYTE_END_TRANSFER:
++            u8g2->gpio_and_delay_cb(u8g2, U8X8_MSG_DELAY_NANO, u8g2->display_info->pre_chip_disable_wait_ns, NULL);
++            u8x8_gpio_SetCS(u8g2, u8g2->display_info->chip_disable_level);
++
++            spi_release(dev);
++            break;
++        default:
++            return 0;
++    }
++
++    return 1;
++}
++#endif /* SPI_NUMOF */
++
++#if I2C_NUMOF
++uint8_t u8x8_byte_riotos_hw_i2c(u8x8_t *u8g2, uint8_t msg, uint8_t arg_int, void *arg_ptr)
++{
++    static uint8_t buffer[255];
++    static uint8_t index;
++
++    i2c_t dev = (i2c_t) u8g2->dev;
++
++    switch (msg) {
++        case U8X8_MSG_BYTE_SEND:
++            while (arg_int--) {
++                buffer[index++] = *((uint8_t *)arg_ptr++);
++            }
++            break;
++        case U8X8_MSG_BYTE_INIT:
++            i2c_init_master(dev, I2C_SPEED_FAST);
++            break;
++        case U8X8_MSG_BYTE_SET_DC:
++            break;
++        case U8X8_MSG_BYTE_START_TRANSFER:
++            i2c_acquire(dev);
++            index = 0;
++            break;
++        case U8X8_MSG_BYTE_END_TRANSFER:
++            i2c_write_bytes(dev, u8x8_GetI2CAddress(u8g2), (char *) buffer, index);
++            i2c_release(dev);
++            break;
++        default:
++            return 0;
++    }
++
++    return 1;
++}
++#endif /* I2C_NUMOF */
+diff --git a/csrc/u8x8.h b/csrc/u8x8.h
+index 1a8fff9..d3c4022 100644
+--- a/csrc/u8x8.h
++++ b/csrc/u8x8.h
+@@ -107,6 +107,8 @@
+ #include <stdint.h>
+ #include <stddef.h>
+ 
++#include "periph/gpio.h"
++
+ #if defined(__GNUC__) && defined(__AVR__)
+ #include <avr/pgmspace.h>
+ #endif 
+@@ -154,9 +156,9 @@ extern "C" {
+ #  define u8x8_pgm_read(adr) (*(const uint8_t *)(adr)) 
+ #endif
+ 
+-#ifdef ARDUINO
+-#define U8X8_USE_PINS
+-#endif
++//#ifdef ARDUINO
++//#define U8X8_USE_PINS
++//#endif
+ 
+ /*==========================================*/
+ /* U8X8 typedefs and data structures */
+@@ -308,6 +310,10 @@ struct u8x8_struct
+ #ifdef U8X8_USE_PINS 
+   uint8_t pins[U8X8_PIN_CNT];	/* defines a pinlist: Mainly a list of pins for the Arduino Envionment, use U8X8_PIN_xxx to access */
```

Comment:
Seems that `git format-patch` share options with `git diff` : see [this page](https://git-scm.com/docs/git-format-patch), so `git format-patch -n sha1 --unified=0` should work.